install/installer/pkg/config/v1/envvars.go

```diff
@@ -0,0 +1,418 @@
+	SSHGatewayHostKeyName                string   `env:"SSH_GATEWAY_HOST_KEY_NAME"`
+	StorageProvider                      string   `env:"STORE_PROVIDER" envDefault:"incluster"`
+	StorageRegion                        string   `env:"STORE_REGION"`
+	StorageAzureCredsName                string   `env:"STORE_AZURE_CREDENTIALS_NAME"`
+	StorageGCPProjectName                string   `env:"STORE_GCP_PROJECT"`
+	StorageGCPServiceAccountName         string   `env:"STORE_GCP_SERVICE_ACCOUNT_NAME"`
+	StorageS3Bucket                      string   `env:"STORE_S3_BUCKET"`
+	StorageS3CredsName                   string   `env:"STORE_S3_CREDENTIALS_NAME"`
+	StorageS3Endpoint                    string   `env:"STORE_S3_ENDPOINT"`
+	TLSSelfSignedEnabled                 bool     `env:"TLS_SELF_SIGNED_ENABLED"`
+	TLSCertManagerEnabled                bool     `env:"CERT_MANAGER_ENABLED"`
+	TLSCustomCACertEnabled               bool     `env:"TLS_CUSTOM_CA_CRT_ENABLED"`
+	TLSCustomCACertCredsName             string   `env:"TLS_CUSTOM_CA_CRT_CREDENTIALS_NAME"`
+	UserManagementBlockEnabled           bool     `env:"USER_MANAGEMENT_BLOCK_ENABLED"`
+	UserManagementBlockPassList          []string `env:"USER_MANAGEMENT_BLOCK_PASSLIST"`
+}
+
+func (c *ConfigEnvvars) load() {
+	t := reflect.TypeOf(*c)
+	v := reflect.ValueOf(c).Elem()
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		tag := field.Tag.Get("env")
+		defaultTag := field.Tag.Get("envDefault")
+
+		envvar, ok := os.LookupEnv(tag)
+		if !ok {
+			envvar = defaultTag
+		}
+
+		valueField := v.Field(i)
+		if valueField.IsValid() && valueField.CanSet() {
+			switch valueField.Kind() {
+			case reflect.Bool:
+				valueField.SetBool(isTruthy(envvar))
+			case reflect.Slice:
+				// This only currently works for slices of strings
+				sliceValue := strings.Split(envvar, " ")
+				slice := reflect.MakeSlice(reflect.SliceOf(field.Type.Elem()), 0, len(sliceValue))
+
+				for _, i := range sliceValue {
+					// Clean up any whitespace and append to slice
+					slice = reflect.Append(slice, reflect.ValueOf(strings.TrimSpace(i)))
+				}
+
+				valueField.Set(slice)
+			default:
+				valueField.SetString(envvar)
+			}
+		}
+	}
+}
+
+type dockerConfigJson struct {
+	Auths map[string]interface{} `json:"auths"`
+}
+
+// BuildFromEnvvars complete the configuration based on envvars - this may remove existing values
+// **IMPORTANT** this function does not guarantee validity of the generated configuration
+func (v version) BuildFromEnvvars(in interface{}) error {
+	cfg, ok := in.(*Config)
+	if !ok {
+		return config.ErrInvalidType
+	}
+
+	envvars := ConfigEnvvars{}
+	envvars.load()
+
+	log.Infof("Detected envvars: %+v", envvars)
+
+	cfg.Domain = envvars.Domain
+
+	if licenseName := envvars.LicenseName; licenseName != "" {
+		log.Infof("Setting license name: %s", licenseName)
+		cfg.License = &ObjectRef{
+			Kind: ObjectRefSecret,
+			Name: licenseName,
+		}
+	} else {
+		cfg.License = nil
```

Comment:
So, I'm wondering why we have this else clause. Considering that we have `License` marked with the `omitempety` tag with a `*` type, This should be `nil` already right? when we are not setting.

Reply:
The reason I've done this is because it may be run multiple times. The first time the license envvar was detected, the second it was not - if we didn't have the `else`, then the value wouldn't be unset.

Granted, this is an edge case, but it is logically possible. Throughout this function, it explicitly sets things to the default values.